exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetComparisonPredicate.java

```diff
@@ -83,23 +85,29 @@ private ParquetComparisonPredicate(
    * where Column1 and Column2 are from same parquet table.
    */
   @Override
-  public boolean canDrop(RangeExprEvaluator<C> evaluator) {
+  public RowsMatch matches(RangeExprEvaluator<C> evaluator) {
     Statistics<C> leftStat = left.accept(evaluator, null);
     if (isNullOrEmpty(leftStat)) {
-      return false;
+      return RowsMatch.SOME;
     }
-
     Statistics<C> rightStat = right.accept(evaluator, null);
     if (isNullOrEmpty(rightStat)) {
-      return false;
+      return RowsMatch.SOME;
     }
-
-    // if either side is ALL null, = is evaluated to UNKNOWN -> canDrop
     if (isAllNulls(leftStat, evaluator.getRowCount()) || isAllNulls(rightStat, evaluator.getRowCount())) {
-      return true;
+      return RowsMatch.NONE;
+    }
+    if (leftStat.hasNonNullValue() && !leftStat.hasNonNullValue() || rightStat.hasNonNullValue() && !rightStat.hasNonNullValue()) {
```

Comment:
When does the condition evaluate to `true`?

Reply:
Yes I noticed but the commit was already pushed and I had to wait for the tests (45 mn).
Corrected in the new push.

Reply:
Please do your own code review and address all review concerns and let me know when the PR is ready for the code review.